src/otx/core/data/dataset/detection.py

```diff
@@ -0,0 +1,67 @@
+"""Module for OTXDetectionDataset."""
+
+from __future__ import annotations
+
+from typing import Callable, Optional
+
+import cv2
+import numpy as np
+import torch
+from datumaro import Bbox, DatasetSubset, Image
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.detection import DetBatchDataEntity, DetDataEntity
+
+from .base import OTXDataset, Transforms
+
+
+class OTXDetectionDataset(OTXDataset[DetDataEntity]):
+    """OTXDataset class for detection task."""
+
+    def __init__(self, dm_subset: DatasetSubset, transforms: Transforms) -> None:
+        super().__init__(dm_subset, transforms)
+
+    def _get_item_impl(self, index: int) -> Optional[DetDataEntity]:
+        item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
+
+        img = item.media_as(Image)
+        img_data = img.data
+        # TODO(vinnamkim): This is a temporal approach
+        # There is an upcoming Datumaro patch here for this
+        # https://github.com/openvinotoolkit/datumaro/pull/1194
+        if img_data.shape[-1] == 4:
+            img_data = cv2.cvtColor(img_data, cv2.COLOR_BGRA2BGR)
+        img_shape = img.size
+
+        bbox_anns = [ann for ann in item.annotations if isinstance(ann, Bbox)]
+
+        bboxes = (
+            np.stack([ann.points for ann in bbox_anns], axis=0).astype(np.float32)
+            if len(bbox_anns) > 0
+            else np.zeros((0, 4), dtype=np.float32)
+        )
+
+        entity = DetDataEntity(
+            image=img_data,
+            img_info=ImageInfo(
```

Comment:
Do we need a ImageInfo data class to present meta info or can we just use dictionary?

Reply:
It is about the super important principle I proposed. Please see the page 17 at my slides. There will be no unstructured data object (e.g., Python dictionary) in OTX v2 to make it developer friendly as possible. This principle will not be changed.